File: javascript/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let indices = {};
  for (let i = 0; i < nums.length; i++) {
    let diff = target - nums[i];
    if (diff in indices) {
      return [i, indices[diff]];
    } else {
      indices[nums[i]] = i;
    }
  }
};

File: python/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indices = {}
        for i, num in enumerate(nums):
            diff = target - num
            if diff in indices:
                return [i, indices[diff]]
            else:
                indices[diff] = i